src/poisson.f90

```diff
@@ -88,7 +88,7 @@ subroutine poisson_xxx(rhs)
        real(mytype), dimension(:,:,:), intent(inout) :: rhs
      end subroutine poisson_xxx
   end interface
-  procedure (poisson_xxx), pointer :: poisson
+  procedure (poisson_xxx), pointer :: poisson=>null()
```

Comment:
ah - might want to be careful here, this may have an implicit ``SAVE`` attribute (which may or may not be desired / affect things)

Reply:
I suspect we want the save attribute for this pointer ? Without it, one can not call the poisson subroutine...

Reply:
ah yes - the danger of looking at small segments of code. It was more to draw attention than say "this is wrong" - I have been caught out by this in subroutine/function definitions before.